index/segment/mem/searcher.go

```diff
@@ -21,37 +21,70 @@
 package mem
 
 import (
+	"fmt"
+	"sort"
+
 	"github.com/m3db/m3ninx/doc"
 	"github.com/m3db/m3ninx/index/segment"
 )
 
+var (
+	errFirstFilterMustNotBeNegation = fmt.Errorf("first filter must be non-negation")
+)
+
+type negationMergeFn func(x segment.PostingsList, y segment.ImmutablePostingsList)
+
+func differenceNegationFn(x segment.PostingsList, y segment.ImmutablePostingsList) { x.Difference(y) }
+func unionNegationFn(x segment.PostingsList, y segment.ImmutablePostingsList)      { x.Union(y) }
```

Comment:
When would this be used? My initial reaction is that we would only ever want to take the difference when we get a negation.